rdmd.d

```diff
@@ -59,6 +59,15 @@ version(unittest) {} else
 int main(string[] args)
 {
     //writeln("Invoked with: ", args);
+    string binPath = thisExePath();
+    if (binPath.exists && binPath.isFile && filenameCmp(baseName(binPath),
+                                                        baseName(args[0])) == 0)
```

Comment:
I don't know what you mean about the extension, currently `compiler` does nothing with extensions.  It simply passes `dmd`, `ldmd2`, or `gdmd` to `spawnProcess`.